### PR TITLE
=doc Consolidate Client Side Request Level docs #1290

### PR DESCRIPTION
**Pull request by jlprat**
Issue: #1290

### CODE REVIEW

File: docs/src/test/java/docs/http/javadsl/SingleRequestInActorExample.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+public class SingleRequestInActorExample extends AbstractActor {
+    final Http http = Http.get(context().system());
+    final ExecutionContextExecutor dispatcher = context().dispatcher();
+    final Materializer materializer = ActorMaterializer.create(context());
+
+    public SingleRequestInActorExample() { // syntax changes slightly in Akka 2.5, see the migration guide
+        receive(ReceiveBuilder
+                .match(String.class, url -> {
+                    pipe(fetch(url), dispatcher).to(self());
+                }).build());
```
Comment (jonas) [inline docs/src/test/java/docs/http/javadsl/SingleRequestInActorExample.java]: The formatting looks a bit weird.

File: docs/src/main/paradox/scala/http/client-side/request-level.md [hunk 2]
```diff
@@ -13,40 +13,47 @@ from a background with non-"streaming first" HTTP Clients.
 @@@ note
 The request-level API is implemented on top of a connection pool that is shared inside the ActorSystem. A consequence of
```
Comment (jonas) [inline docs/src/main/paradox/scala/http/client-side/request-level.md]: `s/ActorSystem/actor system/`?

File: docs/src/main/paradox/scala/http/client-side/request-level.md [hunk 3]
```diff
@@ -13,40 +13,47 @@ from a background with non-"streaming first" HTTP Clients.
...
+:   @@snip [HttpClientExampleDocTest.java](../../../../../test/java/docs/http/javadsl/HttpClientExampleDocTest.java) { #single-request-example }
 
 ### Using the Future-Based API in Actors
 
-When using the `Future` based API from inside an `Actor`, all the usual caveats apply to how one should deal
-with the futures completion. For example you should not access the Actors state from within the Future's callbacks
-(such as `map`, `onComplete`, ...) and instead you should use the `pipeTo` pattern to pipe the result back
-to the Actor as a message.
+When using the @scala[`Future`]@java[`CompletionStage`] based API from inside an `Actor`, all the usual caveats apply to how one should deal
+with the futures completion. For example you should not access the Actors state from within the @scala[`Future`]@java[`CompletionStage`]'s callbacks
```
Comment (jonas) [inline docs/src/main/paradox/scala/http/client-side/request-level.md]: > ... access the actor's state ...

or

> ... access the `Actor`'s state ..

File: docs/src/main/paradox/scala/http/client-side/request-level.md [hunk 4]
```diff
@@ -13,40 +13,47 @@ from a background with non-"streaming first" HTTP Clients.
...
+(such as `map`, `onComplete`, ...) and instead you should use the @scala[`pipeTo`]@java[`pipe`] pattern to pipe the result back
+to the Actor as a message:
```
Comment (jonas) [inline docs/src/main/paradox/scala/http/client-side/request-level.md]: `... to the actor as a message`? Or add backticks around `Actor`.

File: docs/src/main/paradox/scala/http/client-side/request-level.md [hunk 5]
```diff
@@ -13,40 +13,47 @@ from a background with non-"streaming first" HTTP Clients.
...
-@@snip [HttpClientExampleSpec.scala](../../../../../test/scala/docs/http/scaladsl/HttpClientExampleSpec.scala) { #single-request-in-actor-example }
+Scala
+:   @@snip [HttpClientExampleSpec.scala](../../../../../test/scala/docs/http/scaladsl/HttpClientExampleSpec.scala) { #single-request-in-actor-example }
 
+Java
+:   @@snip [SingleRequestInActorExample.java](../../../../../test/java/docs/http/javadsl/SingleRequestInActorExample.java) { #single-request-in-actor-example }
 
 @@@ warning
 
 Be sure to consume the response entities `dataBytes:Source[ByteString,Unit]` by for example connecting it
```
Comment (jonas) [inline docs/src/main/paradox/scala/http/client-side/request-level.md]: `dataBytes:Source[ByteString,Unit]` is Scala-ish. Maybe

> Always make sure you consume the response entity streams by for example connecting it to a `Sink` ...

File: docs/src/main/paradox/scala/http/client-side/request-level.md [hunk 6]
```diff
@@ -13,40 +13,47 @@ from a background with non-"streaming first" HTTP Clients.
...
 or an extra pool just for the long-running connection instead.
 @@@
 
 ## Future-Based Variant
 
-Most often HTTP client needs are very basic. You simply need the HTTP response for a certain request and don't
+Most often, your HTTP client needs are very basic. You simply need the HTTP response for a certain request and don't
 want to bother with setting up a full-blown streaming infrastructure.
 
 For these cases Akka HTTP offers the `Http().singleRequest(...)` method, which simply turns an `HttpRequest` instance
```
Comment (jonas) [inline docs/src/main/paradox/scala/http/client-side/request-level.md]:
```
... offers the @scala[`Http().singleRequest(...)`]@java[`Http.get(system).singleRequest(...)`] method ...
```

File: docs/src/main/paradox/scala/http/client-side/request-level.md [hunk 7]
```diff
@@ -66,7 +73,7 @@ It creates a new "super connection pool flow", which routes incoming requests to
 depending on their respective effective URIs.
 
 The `Flow` returned by `Http().superPool(...)` is very similar to the one from the @ref[Host-Level Client-Side API](host-level.md), so the
```
Comment (jonas) [inline docs/src/main/paradox/scala/http/client-side/request-level.md]:
```
... by @scala[`Http().superPool(...)`]@java[`Http.get(system).superPool(...)`]
```

File: docs/src/main/paradox/scala/http/client-side/request-level.md [hunk 8]
```diff
@@ -66,7 +73,7 @@ It creates a new "super connection pool flow", which routes incoming requests to
 depending on their respective effective URIs.
```
Comment (jonas) [inline docs/src/main/paradox/scala/http/client-side/request-level.md]: One line above
```
... by the @scala[`Http().superPool(...)`]@java[`Http.get(system).superPool(...)`] ...
```

File: docs/src/main/paradox/scala/http/client-side/request-level.md (hunk 6 above)
Comment (jlprat) [inline docs/src/main/paradox/scala/http/client-side/request-level.md]: Agree

File: docs/src/main/paradox/scala/http/client-side/request-level.md (hunk 2 above)
Comment (jlprat) [inline docs/src/main/paradox/scala/http/client-side/request-level.md]: Yes, makes sense

File: docs/src/main/paradox/scala/http/client-side/request-level.md (hunk 3 above)
Comment (jlprat) [inline docs/src/main/paradox/scala/http/client-side/request-level.md]: :+1:

File: docs/src/main/paradox/scala/http/client-side/request-level.md (hunk 5 above)
Comment (jlprat) [inline docs/src/main/paradox/scala/http/client-side/request-level.md]: I'll prefix it with `@scala` and `@java`

File: docs/src/main/paradox/scala/http/client-side/request-level.md [hunk 9]
```diff
@@ -66,7 +73,7 @@ It creates a new "super connection pool flow", which routes incoming requests to
...
 
 The `Flow` returned by `Http().superPool(...)` is very similar to the one from the @ref[Host-Level Client-Side API](host-level.md), so the
-@ref[Using a Host Connection Pool](host-level.md#using-a-host-connection-pool) section also applies here.
+@scala[@ref[Using a Host Connection Pool](host-level.md#using-a-host-connection-pool)]@java[@ref[Using a Host Connection Pool](host-level.md#using-a-host-connection-pool-java)] section also applies here.
```
Comment (jlprat) [inline docs/src/main/paradox/scala/http/client-side/request-level.md]: Another PR unifies this page. Maybe I can just drop the suffix "java" here and the other PR will do the unification

File: docs/src/main/paradox/scala/http/client-side/request-level.md (hunk 7 above)
Comment (jlprat) [inline docs/src/main/paradox/scala/http/client-side/request-level.md]: Agree

File: docs/src/test/java/docs/http/javadsl/SingleRequestInActorExample.java (hunk 1 above)
Comment (jlprat) [inline docs/src/test/java/docs/http/javadsl/SingleRequestInActorExample.java]: Same format as the in-lined example, let me try to improve it.

File: docs/src/main/paradox/scala/http/client-side/request-level.md (hunk 4 above)
Comment (jlprat) [inline docs/src/main/paradox/scala/http/client-side/request-level.md]: :+1:

File: docs/src/main/paradox/scala/http/client-side/request-level.md (hunk 8 above)
Comment (jlprat) [inline docs/src/main/paradox/scala/http/client-side/request-level.md]: right

**Pull request merged**